src/course-unit/course-sequence/hooks.js

```diff
@@ -0,0 +1,139 @@
+  }
+
+  const sequenceIndex = sequenceIds.indexOf(currentSequenceId);
+  const unitIndex = sequence.unitIds.indexOf(currentUnitId);
+
+  const isFirstSequence = sequenceIndex === 0;
+  const isFirstUnitInSequence = unitIndex === 0;
+  const isFirstUnit = isFirstSequence && isFirstUnitInSequence;
+  const isLastSequence = sequenceIndex === sequenceIds.length - 1;
+  const isLastUnitInSequence = unitIndex === sequence.unitIds.length - 1;
+  const isLastUnit = isLastSequence && isLastUnitInSequence;
+
+  const nextSequenceId = sequenceIndex < sequenceIds.length - 1 ? sequenceIds[sequenceIndex + 1] : null;
+  const previousSequenceId = sequenceIndex > 0 ? sequenceIds[sequenceIndex - 1] : null;
+
+  let nextLink;
+  const nextIndex = unitIndex + 1;
+
+  if (nextIndex < sequence.unitIds.length) {
+    const nextUnitId = sequence.unitIds[nextIndex];
+    nextLink = `/course/${courseId}/container/${nextUnitId}/${currentSequenceId}`;
+  } else if (nextSequenceId) {
+    const nextUnitId = decodeURIComponent(nextUrl);
+    nextLink = `/course/${courseId}${nextUnitId}/${nextSequenceId}`;
+  }
+
+  let previousLink;
+  const previousIndex = unitIndex - 1;
+
+  if (previousIndex >= 0) {
+    const previousUnitId = sequence.unitIds[previousIndex];
+    previousLink = `/course/${courseId}/container/${previousUnitId}/${currentSequenceId}`;
+  } else if (previousSequenceId) {
+    const previousUnitId = decodeURIComponent(prevUrl);
+    previousLink = `/course/${courseId}${previousUnitId}/${previousSequenceId}`;
+  }
+
+  return {
+    isFirstUnit, isLastUnit, nextLink, previousLink,
+  };
+}
+
+const invisibleStyle = {
+  position: 'absolute',
+  left: 0,
+  pointerEvents: 'none',
+  visibility: 'hidden',
+};
+
+/**
+ * This hook will find the index of the last child of a containing element
+ * that fits within its bounding rectangle. This is done by summing the widths
+ * of the children until they exceed the width of the container.
+ *
+ * The hook returns an array containing:
+ * [indexOfLastVisibleChild, containerElementRef, invisibleStyle, overflowElementRef]
+ *
+ * indexOfLastVisibleChild - the index of the last visible child
+ * containerElementRef - a ref to be added to the containing html node
+ * invisibleStyle - a set of styles to be applied to child of the containing node
+ *    if it needs to be hidden. These styles remove the element visually, from
+ *    screen readers, and from normal layout flow. But, importantly, these styles
+ *    preserve the width of the element, so that future width calculations will
+ *    still be accurate.
+ * overflowElementRef - a ref to be added to an html node inside the container
+ *    that is likely to be used to contain a "More" type dropdown or other
+ *    mechanism to reveal hidden children. The width of this element is always
+ *    included when determining which children will fit or not. Usage of this ref
+ *    is optional.
+ */
+export function useIndexOfLastVisibleChild() {
```

Comment:
This exists in Paragon, maybe we can import this one from Paragon?

Reply:
I was unable to get an identical visual result when importing this hook from Paragon. Now let’s not change what works in MFE Learning. But in the future, it will be a good update!